src/components/ContactUsForm/ContactUs.jsx

```diff
@@ -1,82 +1,332 @@
+import Supply_6 from './Supply_6.png';
+import Supply_7 from './Supply_7.png';
 
 const ContactUs = () => {
-  const business_ID = 0;
+  const FPH_ID = 0;
   const { backend } = useBackend();
-  const [name, setName] = useState('business?'); // name of business
+  const { currentUser } = useAuth();
+  const navigate = useNavigate();
   const [text, setText] = useState('');
-  const [checkedItems, setCheckedItems] = useState([false, false, false, false, false]);
-  const formData = {
-    business_id: business_ID,
-    message: null,
-    timestamp: new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angeles' }),
-    been_dismissed: false,
-  };
+  const [checkedItems, setCheckedItems] = useState([0, 0, 0, 0, 0, 0, 0, 0]);
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const [isFormValid, setIsFormValid] = useState(false);
+  const checkedThingies = [
+    'Get Pet Food Decal',
+    'Decal',
+    'Homeless Card',
+    'Business Card',
+    'Donation Site Decal',
+    'Donation Site Bin Decals',
+    'Donation Envelopes',
+    'Homeless Card 2',
+  ];
+  const supplyDescriptions = [
+    'Stickers to share with your Pet Food Provider.',
+    'Stickers that can be handed out and can be placed on water bottles, cars and more.',
+    'Cards to share with your Pet Food Provider who may have more direct contact with the homeless in need of emergency veterinary services.',
+    'Feeding Pets of the Homeless’ contact information cards.',
+    'Decals that indicate that you are a donation site.',
+    'Decorate your donation bins with these decals so donators know where to put their items. ',
+    'Envelopes to provide potential donators to make a direct donation.',
+    'Cards to share with your Pet Food Provider who may have more direct contact with the homeless who are in need of emergency veterinary services.',
+  ];
+  const suppliesImages = [
+    Supply_0,
+    Supply_1,
+    Supply_2,
+    Supply_3,
+    Supply_4,
+    Supply_5,
+    Supply_6,
+    Supply_7,
+  ];
+
+  const [businessId, setBusinessId] = useState(null);
+  const [businessName, setBusinessName] = useState('');
+
+  useEffect(() => {
+    // Define an async function inside useEffect since useEffect cannot be asynchronous directly
+    const fetchBusinessId = async () => {
+      try {
+        // Fetch business ID from backend
+        const businessIdResponse = await backend.get(`/businessUser/${currentUser.uid}`);
+        const fetchedBusinessId = businessIdResponse.data[0].id;
+        const businessNameResponse = await backend.get(`/business/${fetchedBusinessId}`);
+        const fetchedBusinessName = businessNameResponse.data[0].name;
+        // Set the fetched business ID in state
+        setBusinessId(fetchedBusinessId);
+        setBusinessName(fetchedBusinessName);
+      } catch (error) {
+        // Handle errors
+        console.error('Error fetching business ID:', error);
+      }
+    };
+
+    fetchBusinessId();
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'backend' and 'currentUser.uid'. Either include them or remove the dependency array.